charms/worker/k8s/src/containerd.py

```diff
@@ -278,48 +282,39 @@ def ensure_hosts_toml(self):
         _ensure_file(hosts_toml_path, tomli_w.dumps(self.hosts_toml), 0o600, 0, 0)
 
 
-class RegistryConfigs(pydantic.BaseModel, extra=pydantic.Extra.forbid):
-    """Represents a set of containerd registries.
-
-    Attrs:
-        registries (List[Registry]):
-    """
+Registries = List[Registry]
+registry_list = pydantic.TypeAdapter(Registries)
 
-    registries: List[Registry]
 
-
-def parse_registries(json_str: str) -> List[Registry]:
+def parse_registries(yaml_str: str) -> Registries:
     """Parse registry configurations from json string.
 
     Args:
-        json_str (str): raw user supplied content
+        yaml_str (str): raw user supplied content
 
     Returns:
-        RegistryConfigs parsed from json_str
+        List[Registries] parsed from yaml_str
 
     Raises:
         ValueError: if configuration is not valid
     """
-    if not json_str:
-        return []
-
     try:
-        parsed = json.loads(json_str)
-    except json.JSONDecodeError as e:
-        raise ValueError(f"not valid JSON: {e}") from e
+        parsed = yaml.safe_load(yaml_str or "[]")
+    except yaml.error.YAMLError as e:
+        raise ValueError(f"not valid YAML: {e}") from e
```

Comment:
NEAT!  json is natively handled by a yaml parser.  But now yaml is supported too..